pkg/drivers/sqlite/sqlite.go

```diff
@@ -105,3 +118,76 @@ func setup(db *sql.DB) error {
 
 	return nil
 }
+
+// alterTableIndices drops the given old table indices from the existing
+// kine table (if it exists, and if the indices exists), and then creates
+// the given new ones on the kine table.
+func alterTableIndices(d *generic.Generic) error {
+	if d.LockWrites {
+		d.Lock()
+		defer d.Unlock()
+	}
+
+	// Drop the old indices
+	for _, stmt := range dropIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create the new indices
+	for _, stmt := range createIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkMigrate performs migration from an old key value table to the kine
+// table only if the old key value table exists and migration has not been
+// done already.
+func checkMigrate(ctx context.Context, d *generic.Generic) error {
```

Comment:
I think we need a better name. `checkMigrate` makes the reader think that it will "check" only, while this is actually the migration code. Maybe `versionedMigrate`? `performMigration`? Or any other thing that clearly states that a migration is happening